docs/lightning-message-channels.md

```diff
@@ -0,0 +1,147 @@
+
+(c-lightning's [plugin architecture](https://lightning.readthedocs.io/PLUGINS.html) is a big part of what makes this relatively easy to implement; it means we have a very loose coupling via a simple plugin that forwards messages and notifications from the c-lightning node to joinmarket's jmdaemon, which is able to use it as "just another way of passing messages", basically).
+
+What is in this PR *so far* as of 31 Aug 2021, time of writing, is limited, but functional. The directory nodes are queried for connections (after being connected to, of course), to give takers a view of the current trading pit. They also basically do all the heavy lifting, broadcasting `pubmsgs` and are used to forward `privmsgs` also (except in cases where participants already have an existing connection). This will be changed at some point to use onion routing with hops, which is already included in the `sendonionmessage` API, see [here](https://lightning.readthedocs.io/lightning-sendonionmessage.7.html) for details - notice the `hops` array. I say *will be changed*, because it's really needed if we want to gain the scale that such a more decentralized approach can achieve.
+
+
+<a name="install-ln" />
+
+## C-Lightning installation.
+
+Obviously it's a disadvantage to require an additional piece of software (if you're not already running it, or if you're running lnd), but there are two ameliorating factors:
+
+* You don't, for now, need to even fund the wallet or use anything in the software apart from running the Joinmarket plugin.
+
+* It's actually very easy to build and install on Linux.
+
+The installation instructions are on the repo, however there is a very important detail before you go there, and follow those instructions. You unfortunately *must* compile and build, not just install the binaries, because `sendonionmessage` is still in the "experimental features" set, and you must run:
+
+```
+./configure --enable-developer --enable-experimental-features`
+```
+
+and not just
+
+```
+./configure
+```
+
+With that said, follow the guide for building on your Linux distro [here](https://github.com/ElementsProject/lightning/blob/master/doc/INSTALL.md).
+
+
+<a name="regtest" />
+
+## Running tests on regtest
+
+Since this doc (and code) are in a testing phase, you will want to run on either regtest or signet to start. Here there's an especially good reason to get it running on regtest: c-lightning has a purpose built environment to make this very easy. Once you've completed the installation instructions, and assuming you have `bitcoind` and `bitcoin-cli` in your path, you'll be able to:
```

Comment:
Technically you can also use testnet3 in addition to regtest and signet.

Reply:
True, I just don't want to advise it, I think it's not a helpful network any more.